API/utils/wildcard.go

```diff
@@ -0,0 +1,33 @@
+package utils
+
+import (
+	"regexp"
+	"strings"
+
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+const NAME_CHARACTER_REGEX = `(\w|\-)*`                              // accepted characters to compose ids
+const NAME_REGEX = `\w` + NAME_CHARACTER_REGEX                       // accepted regex for names that compose ids
+const NAME_RECURSIVE_REGEX = `(\.` + NAME_REGEX + `)*\.`             // accepted regex for names that compose ids in recursive way
+const NAME_RECURSIVE_REGEX_WITH_DEPTH = `(\.` + NAME_REGEX + `)$1\.` // accepted regex for names that compose ids in recursive way with min and max depth
+
+var doubleStarWithDepthRegex = regexp.MustCompile(`\\\.\*\*({\d+,\d*})\\\.`) // ** with min and max depth
+var doubleStarRegex = regexp.MustCompile(`\\\.\*\*\\\.`)                     // ** without depth
+var pointStar = ".*"
+var starRegex = regexp.MustCompile(`([^\)]|^)\*+`) // * with something different to ")" before to avoid replacing the * written in the previous steps
+
+func applyWildcards(value string) string {
```

Comment:
If I send id=BASIC.A.R1.** it will return BASIC\.A\.R1\.\w(\w|\-)*(\w|\-)* which will give me the same result as sending BASIC.A.R1.* (returns BASIC\.A\.R1\.\w(\w|\-)*), that is, only direct children.